### PR TITLE
Bugfix/checkbox validation design

### LINKED ISSUE #1572: Checkbox validation design is not equal anymore
The checkbox doesn't look correct anymore in its invalid state:

Please also consider hover states.

IS:
Screenshot 2019-12-18 at 14 34 56

SHOULD:
Screenshot 2019-12-18 at 14 35 15

SHOULD:
Screenshot 2019-12-18 at 14 36 46

And I guess this is also not valid. This state should only exist in hover-state, but this would have to be checked against the design guidelines or with the designers:
Screenshot 2019-12-18 at 14 54 24

### PR DESCRIPTION
**Pull request by 01231**
Fixes #1572.
Design: [zeplin](https://app.zeplin.io/project/5bed40045544f67ec5c16d69/screen/5ca617fd47f2319f5d7a6155)
# Done is when (DoD):
- [ ] I have added UI/Unit tests for my changes
- [ ] I added modified component properties to the typescript typings
- [x] Design has been reviewed with @kenaxa 
- [x] My changes are accordingly to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)
- [x] Old tests and eslint rule are not broken
- [ ] My changes are well documented in the README and showcased in the stories 
- [ ] Tested with IE

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas (focus on **WHY**)
- [x] I have made/updated the ChangeLog Section in the README

### CODE REVIEW

**Comment by raphaellueckl** [comment]
If the checkbox is invalid BUT selected, the border around it is no longer red. I guess that is a new issue:

Feature-Branch deployment:
https://axa-ch.github.io/plib-feature/bugfix/checkbox-validation-design/iframe.html?id=atoms-checkbox-react--checkbox-uncontrolled&knob-label=this%20is%20a%20labelasd&knob-name=my-checkboxasd&knob-checked=true&knob-error=true&knob-required=true

Screenshot 2019-12-20 at 12 50 27


Prod:
https://patterns.axa.ch/iframe.html?id=atoms-checkbox-react--checkbox-uncontrolled&knob-label=this%20is%20a%20labelasd&knob-name=my-checkboxasd&knob-checked=true&knob-error=true&knob-required=true

Screenshot 2019-12-20 at 12 50 33

File: src/components/10-atoms/checkbox/react/DemoCheckboxLabelAsChildren.jsx [hunk 1]
```diff
@@ -10,7 +10,17 @@ const DemoCheckboxLabelAsChildren = () => {
 
   return (
     <AXACheckbox checked={checked} onChange={handleCheckboxChange}>
-      <p>or click here if you prefere</p>
+      <p>
+        click{' '}
```
Comment (raphaellueckl) [inline src/components/10-atoms/checkbox/react/DemoCheckboxLabelAsChildren.jsx:14]: What is the {' '} for? Doesn't html insert a whitespace by itself here?

File: src/components/10-atoms/checkbox/ui.test.js [hunk 2]
```diff
@@ -87,7 +87,7 @@ test('should style checked error checkbox-icon inner box correctly', async t =>
   await t
     .wait(50)
     .expect(await getIconBackgroundColor())
-    .eql('rgb(201, 20, 50)');
+    .eql('rgb(0, 0, 143)'); // has to be blue when seleted, due to styleguide specification
```
Comment (raphaellueckl) [inline src/components/10-atoms/checkbox/ui.test.js:90]: The styleguide wants that? If true, my comments with links and screenshots is obsolete.

**Review by raphaellueckl — CHANGES_REQUESTED** [review]
Very nice job! Just some minor ideas from my side.

File: src/components/10-atoms/checkbox/react/DemoCheckboxLabelAsChildren.jsx (hunk 1 above)
Comment (01231) [reply src/components/10-atoms/checkbox/react/DemoCheckboxLabelAsChildren.jsx:14]: No it doesn't for react. Prettier did this.

File: src/components/10-atoms/checkbox/ui.test.js (hunk 2 above)
Comment (01231) [reply src/components/10-atoms/checkbox/ui.test.js:90]: Yes, this is what I found out while chatting with Thomas and looking at https://app.zeplin.io/project/5bed40045544f67ec5c16d69/screen/5ca617fd47f2319f5d7a6155.

Comment (01231) [reply src/components/10-atoms/checkbox/ui.test.js:90]: I also asked ken to approve from the designer side - and he did.

**Review by raphaellueckl — APPROVED** [review]

**Pull request merged**